python/ray/_private/worker.py

```diff
@@ -1499,7 +1499,11 @@ def init(
         namespace=namespace,
         job_config=job_config,
     )
-    if job_config and job_config.code_search_path:
+    if job_config and (
+        job_config.runtime_env.get("py_modules")
+        or job_config.runtime_env.get("java_jars")
+        or job_config.runtime_env.get("native_libraries")
+    ):
         global_worker.set_load_code_from_local(True)
```

Comment:
This load_code_from_local is not complete. We should always disable this one (which might impact the perf, but it won't make things wrong).